tests/ci/integration/run_ruby_integration.sh

```diff
@@ -0,0 +1,85 @@
+#!/bin/bash -exu
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0 OR ISC
+
+source tests/ci/common_posix_setup.sh
+
+set -exuo pipefail
+
+# Set up environment.
+
+# SYS_ROOT
+#  - SRC_ROOT(aws-lc)
+#    - SCRATCH_FOLDER
+#      - RUBY_SRC_FOLDER
+#        - ruby_3_1
+#      - RUBY_PATCH_FOLDER
+#        - ruby_3_1
+#      - AWS_LC_BUILD_FOLDER
+#      - AWS_LC_INSTALL_FOLDER
+
+# Assumes script is executed from the root of aws-lc directory
+SCRATCH_FOLDER="${SRC_ROOT}/RUBY_BUILD_ROOT"
+RUBY_SRC_FOLDER="${SCRATCH_FOLDER}/ruby-src"
+RUBY_PATCH_FOLDER="${SRC_ROOT}/tests/ci/integration/ruby_patch"
+AWS_LC_BUILD_FOLDER="${SCRATCH_FOLDER}/aws-lc-build"
+AWS_LC_INSTALL_FOLDER="${SCRATCH_FOLDER}/aws-lc-install"
+
+function ruby_build() {
+    local branch=${1}
+    pushd ${branch}
```

Comment:
It would be more robust to not assume the current directory, but fully qualify the path:
```
pushd "${RUBY_SRC_FOLDER}/${branch}"
```